doc/groups.md

```diff
@@ -1 +1,573 @@
 # Groups Module
+Module | HTTP Method | URL | Description 
+--- | --- | --- | ---
+[Add Groups](#add) | POST | /groups | Add Groups API
+[Edit Groups](#edit) | PUT | /groups/{id} | Edit Groups API
+[View All Groups](#view) | GET | /groups | View Groups API
+[View Groups Detail](#view-detail) | GET | /groups/{id} | View Detail Group API
+[Delete Groups](#delete) | DELETE | /groups/{id} | Delete Groups API
+
+
+### Database
+![](./groups-layer.png)
+
+
+## <a name="add"></a>Add Groups
+
+### Endpoint 
+POST /groups
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+
+### Request Payloads
+Name | Type | Example Value
+--- | --- | ---
+title | string | Admin  
+description | string | Administrator
+```
+{
+    "title": "Admin",
+    "description": "Administrator"
+}
+
+```
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+500 | Internal Server Error | some un-handle error in server 
+201 | Created | Created
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "Bad Request",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-201",
+    "status_message": "Created",
+    "data": {
+        "id": 1,
+        "title": "Admin",
+        "description": "Administrator",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-01 00:00:00"
+    }
+} 
+```
+
+### Logic
+
```

Comment:
add logic: after save data in groups table, also save the data on aros table 
-   id --> auto increment
- parent_id --> null
- model --> Groups
- foreign_key --> id of group
- alias --> name of group
- lft --> grab using cakephp tree behavior
- rght --> grab using cakephp tree behavior